developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -0,0 +1,2124 @@
+
+[[adding-permission]]
+=== Adding permission
+
+Whether adding a new actions to existing controller or adding a new
+controller, every action must be mapped to a foreman permission. +
+See a typical structure of the security section of the registered plugin
+method:
+
+[source, ruby]
+----
+security_block :security_block_name do
+          permission :view_something, {:controller_name => [:index, :show, :auto_complete_search] }
+          permission :new_something, {:controller_name => [:new, :create] }
+          permission :edit_something, {:controller_name => [:edit, :update] }
+          permission :delete_something, {:controller_name => [:destroy] }
+end
+----
+[[adding-your-permissions-to-foremans-roles]]
+=== Adding your permissions to Foreman's roles
+
+_Requires Foreman 1.15 or higher, set `requires_foreman '>= 1.15'` in
+engine.rb_
+
+Plugins should merge seamlessly with the rest of the application.
+Foreman provides you with several DSL methods to add your permissions to
+existing Foreman's roles. +
+That way, users with these roles have access to your plugin's
+functionality without a need to change anything.
+
+[source, ruby]
+----
+security_block :security_block_name do
+  # define your permissions
+end
+
+# add permissions to Manager and Viewer roles
+add_all_permissions_to_default_roles
+----
+
+If you need more control over what needs to be added you can use the
+following:
+
+[source, ruby]
+----
+add_permissions_to_default_roles 'Manager' => [:first_permission, :second_permission], 'Viewer' => [:third_permission]
+----
+
+Or alternatively:
+
+[source, ruby]
+----
+add_resource_permissions_to_default_roles ['MyPlugin::FirstResource', 'MyPlugin::SecondResource'], :except => [:skip_this_permission]
+----
+
+[[adding-roles]]
+=== Adding roles
+
+The register plugin method allows adding a predefined role, the
+following sample show how to add a role that includes the set of
+permissions from the previous section.
+
+[source, ruby]
+----
+  # Add a new role called 'New Role Name' if it doesn't exist
+  role "New Role Name", [:view_something, :provision_something, :edit_something, :destroy_something]
+----
+[[specifying-alternate-auto-complete-path-for-role-filters]]
+=== Specifying alternate auto-complete path for Role Filters
+
+_Requires Foreman 1.6 or higher, set `requires_foreman '>= 1.6'` in
+engine.rb_
+
+Use search_path_override method with the namespace of your plugin as the
+parameter to define overrides. Usage example:
+
+[source, ruby]
+----
+search_path_override("Katello") do |resource|
+  case resource
+    when 'Katello::Content_View'
+      '/katello/content_views/auto_complete_path'
+    else
+      "katello/#{resource.deconstantise.pluralise}/another_search_path"
+  end
+end
+----
+[[altering-the-menu]]
+=== Altering the menu
+
+A plugin can add menu items, entire sub menus and even delete a menu
+item, here are a few examples:
+
+Adding an item to existing menu:
```

Comment:
Maybe point to the location where the menu code lives in foreman repo? Might be useful for finding out what the current structure is and which `:parent` to use.